services/web/server/src/simcore_service_webserver/users/_notifications_handlers.py

```diff
@@ -33,13 +35,21 @@
 
 
 async def _get_user_notifications(
-    redis_client: aioredis.Redis, user_id: int
+    redis_client: aioredis.Redis, user_id: int, product_name: str
 ) -> list[UserNotification]:
     """returns a list of notifications where the latest notification is at index 0"""
     raw_notifications: list[str] = await redis_client.lrange(
         get_notification_key(user_id), -1 * MAX_NOTIFICATIONS_FOR_USER_TO_SHOW, -1
     )
-    return [UserNotification.parse_raw(x) for x in raw_notifications]
+    notifications = [json.loads(x) for x in raw_notifications]
+    # Make it backwards compatible
+    for n in notifications:
+        if "product" not in n:
+            n["product"] = "UNDEFINED"
```

Comment:
at this point you no longer need this

Reply:
If ``Literal["UNDEFINED"]`` goes first in the model definition, would it be set by default if ``product`` is non existent? When I had it in second position, it wasn't set.